eclipse-collections-api/src/main/java/org/eclipse/collections/api/map/MutableMapIterable.java

```diff
@@ -152,6 +152,25 @@ default boolean removeIf(Predicate2<? super K, ? super V> predicate)
      */
     MutableMapIterable<K, V> withKeyValue(K key, V value);
 
+    /**
+     * This method allows mutable maps the ability to add elements of a Map to their existing
+     * elements. The original map is modified and then returned. In order to use this method properly with mutable maps
+     * the following approach must be taken:
+     * <p>
+     * <pre>
+     * map = map.withMap(someMap);
+     * </pre>
```

Comment:
The javadoc seems wrong. At first I thought this was going to be about a method on mutable/immutable maps. If you're not sure if the method is mutating or not, then the safe thing to do is:
```
newReference = oldReference.possiblyMutatingMethod(args)
```
This one is specifically on mutable maps so the guidance doesn't need to be, and should not be followed. If you just delete "In order to use this method properly" until the end, I think it's good.

Reply:
Roger that!